flink-autoscaler/src/main/java/org/apache/flink/autoscaler/utils/MemoryTuningUtils.java

```diff
@@ -0,0 +1,203 @@
+                                        // Lower limit is the minimum configured heap size
+                                        config.get(AutoScalerOptions.MEMORY_TUNING_MIN_HEAP)
+                                                .getBytes(),
+                                        avgHeapSize.getBytes())));
+        LOG.info("New TM heap memory {}", newHeapSize.toHumanReadableString());
+
+        // Diff can be negative (memory shrinks) or positive (memory grows)
+        final long heapDiffBytes = newHeapSize.getBytes() - maxHeapSize.getBytes();
+
+        final MemorySize totalMemory = adjustTotalTmMemory(context, heapDiffBytes);
+        if (totalMemory.equals(MemorySize.ZERO)) {
+            return config;
+        }
+
+        // Prepare the tuning config for new configuration values
+        var tuningConfig = new Configuration();
+        // Update total memory according to new heap size
+        // Adjust the total container memory and the JVM heap size accordingly.
+        tuningConfig.set(TaskManagerOptions.TOTAL_PROCESS_MEMORY, totalMemory);
+        // Framework and Task heap memory configs add up together yield the max heap memory.
+        // To simplify the calculation, set the framework heap memory to zero.
+        tuningConfig.set(TaskManagerOptions.FRAMEWORK_HEAP_MEMORY, MemorySize.ZERO);
+        tuningConfig.set(TaskManagerOptions.TASK_HEAP_MEMORY, newHeapSize);
+
+        // All memory options which can be configured via fractions need to be set to their
+        // absolute values or, if there is no absolute setting, the fractions need to be
+        // re-calculated.
+        MemorySize managedMemory = memSpecs.getFlinkMemory().getManaged();
+        if (shouldTransferHeapToManagedMemory(config, heapDiffBytes)) {
+            // If RocksDB is configured, give back the heap memory as managed memory to RocksDB
+            MemorySize newManagedMemory =
+                    new MemorySize(managedMemory.getBytes() + Math.abs(heapDiffBytes));
+            LOG.info(
+                    "Increasing managed memory size from {} to {}",
+                    managedMemory,
+                    newManagedMemory);
+            tuningConfig.set(TaskManagerOptions.MANAGED_MEMORY_SIZE, newManagedMemory);
+        } else {
+            tuningConfig.set(TaskManagerOptions.MANAGED_MEMORY_SIZE, managedMemory);
+        }
+
+        tuningConfig.set(
+                TaskManagerOptions.NETWORK_MEMORY_FRACTION,
+                getFraction(
+                        memSpecs.getFlinkMemory().getNetwork(),
+                        new MemorySize(
+                                memSpecs.getTotalFlinkMemorySize().getBytes() + heapDiffBytes)));
+        tuningConfig.set(
+                TaskManagerOptions.JVM_OVERHEAD_FRACTION,
+                getFraction(memSpecs.getJvmOverheadSize(), totalMemory));
+
+        return tuningConfig;
+    }
+
+    private static boolean shouldTransferHeapToManagedMemory(
```

Comment:
As I discussed in the design doc, would you mind introducing a option to control it?